src/docfx/docset/Docset.cs

```diff
@@ -140,9 +149,34 @@ private Docset(Report report, string docsetPath, string locale, Config config, C
             _legacyTemplate = new Lazy<LegacyTemplate>(() =>
             {
                 Debug.Assert(!string.IsNullOrEmpty(Config.Theme));
-                var (themeRemote, branch) = LocalizationUtility.GetLocalizedTheme(Config.Theme, Locale, Config.Localization.DefaultLocale);
-                return new LegacyTemplate(RestoreMap.GetGitRestorePath($"{themeRemote}#{branch}"), Locale);
+                var (themeRemote, themeBranch) = LocalizationUtility.GetLocalizedTheme(Config.Theme, Locale, Config.Localization.DefaultLocale);
+                return new LegacyTemplate(RestoreMap.GetGitRestorePath($"{themeRemote}#{themeBranch}"), Locale);
             });
+
+            _repositories = new ConcurrentDictionary<string, Repository>();
+        }
+
+        public Repository GetRepository(string filePath)
```

Comment:
calling `GitUtility` for each file might be slow, there is a folder level cache in `GitCommitProvider`, maybe reuse that?